reactor-core/src/test/java/reactor/core/publisher/FluxTimeoutTest.java

```diff
@@ -399,4 +399,15 @@ public void timeoutNotDurationMessageSecondTimeout() {
 
 		assertThat(generatorUsed.get()).as("generator used").isTrue();
 	}
+
+	@Test
+	public void onSubscribeRace() {
+		for (int i = 0; i < 10_000; i++) {
+			Flux.just("Hello")
```

Comment:
could use a `StepVerifier` here to validate we go to the fallback all the time ?
also the reporter indicated that a low-millis timeout would sometimes trigger the issue too. maybe we can either add a test with `ofMillis(1)` or set a non-zero timeout for a percentage of the loops (eg. 10%) ?